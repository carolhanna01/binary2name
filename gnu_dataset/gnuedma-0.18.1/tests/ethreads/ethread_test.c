/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */


#include <stdio.h>
#include <edma.h>



int et0_flag = 1;
int et1_flag = 1;

void *
et0_main (ETHREAD_PARAMS p)
{
  OBJID    id;

  edma_thread_register ();
  edma_printf ("[et0] %s", "Initiating et0 Main Loop");

  while (et0_flag)
    {
      edma_printf ("[et0] %s", "et0 working...");

      id = edma_new_obj ("HELLO_WORLD");
      edma_free_obj (id);
      sleep (1);
    }

  edma_thread_unregister();
  return 0;
}


void *
et1_main (ETHREAD_PARAMS p)
{
  edma_thread_register ();
  edma_printf ("[et1] %s", "Initiating et1 Main Loop");

  while (et1_flag)
    {
      edma_printf ("[et1] %s", "et1 working...");
      sleep (1);
    }

  edma_thread_unregister();
  return 0;
}

int 
main (int argc, char *argv[])
{
  OBJID		id, id1;
  ETHREAD       et0, et1;


  EDMAInit();
  edma_printf ("%s", "[Main] Creating Threads....");

  edma_thread_list ();
  sleep (1);
  edma_thread_create (&et0, et0_main, NULL);
  sleep (1);
  edma_thread_list ();
  edma_thread_create (&et1, et1_main, NULL);
  sleep (1);
  edma_thread_list ();

  sleep (6);

  edma_thread_list ();
  edma_printf ("[Main] %s", "Killing Threads...");

  et0_flag = 0;
  sleep (2);
  edma_thread_list ();

  et1_flag = 0;
  sleep (2);
  edma_thread_list ();

  edma_printf ("[Main] %s", "done.");

  EDMAEnd();
}

