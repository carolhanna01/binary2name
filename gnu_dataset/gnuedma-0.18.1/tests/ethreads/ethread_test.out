
GNU/EDMA 0.16.4 build [Jan 30 2010][Beta][0.16.4] starting up...
EDMA Startup process... OK
[Main] Creating Threads....
[edma_thread_list] GNU/EDMA Global Thread List:
[edma_thread_list] Index 0: Thread 0x0xb7de9b60 Thread Stack: 0
[et0] Initiating et0 Main Loop
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[edma_thread_list] GNU/EDMA Global Thread List:
[edma_thread_list] Index 0: Thread 0x0xb7de9b60 Thread Stack: 0
[edma_thread_list] Index 1: Thread 0x0xb7dadb90 Thread Stack: 0
[et1] Initiating et1 Main Loop
[et1] et1 working...
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[et1] et1 working...
[edma_thread_list] GNU/EDMA Global Thread List:
[edma_thread_list] Index 0: Thread 0x0xb7de9b60 Thread Stack: 0
[edma_thread_list] Index 1: Thread 0x0xb7dadb90 Thread Stack: 0
[edma_thread_list] Index 2: Thread 0x0xb75acb90 Thread Stack: 0
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[et1] et1 working...
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[et1] et1 working...
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[et1] et1 working...
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[et1] et1 working...
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[et1] et1 working...
[et0] et0 working...
(HELLO_WORLD)  [Constructor]
(HELLO_WORLD)  [Destructor]
[edma_thread_list] GNU/EDMA Global Thread List:
[edma_thread_list] Index 0: Thread 0x0xb7de9b60 Thread Stack: 0
[edma_thread_list] Index 1: Thread 0x0xb7dadb90 Thread Stack: 0
[edma_thread_list] Index 2: Thread 0x0xb75acb90 Thread Stack: 0
[Main] Killing Threads...
[et1] et1 working...
[edma_thread_unregister] Unregistering thread 0xb7dadb90
======> Index 1 of 3[edma_thread_unregister] Thread 0xb7dadb90 found at index 1
[et1] et1 working...
[edma_thread_list] GNU/EDMA Global Thread List:
[edma_thread_list] Index 0: Thread 0x0xb7de9b60 Thread Stack: 0
[edma_thread_list] Index 1: Thread 0x0xb75acb90 Thread Stack: 0
[edma_thread_unregister] Unregistering thread 0xb75acb90
======> Index 1 of 2[edma_thread_unregister] Thread 0xb75acb90 found at index 1
[edma_thread_list] GNU/EDMA Global Thread List:
[edma_thread_list] Index 0: Thread 0x0xb7de9b60 Thread Stack: 0
[Main] done.
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
